### DISCUSSION TITLE
Martin Jerry SD01 Dimmer (MJ-SD01) Does not Update Dimmer on Manual Button Presses

### DISCUSSION THREAD

**Discussion by abishur** [arendst/Tasmota#16562] [Q&A]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
I have a Martin Jerry SD01 Dimmer (MJ-SD01) (https://templates.blakadder.com/martin_jerry_MJ-SD01.html) that has a toggle button for turning the relay on and off and a 2 button dimmer switch.  Pressing the upper dimmer will cause the status LEDs to increase in brightness and pressing the lower dimmer button will cause the status LEDs to decrease.  However, No MQTT message is sent.  Additionally, using the web interface I can move the dimmer slider back and forth and this will also increase/decrease the status LEDs as appropriate as well as send MQTT messages (i.e. stat/eden_switch/RESULT = {"POWER":"ON","Dimmer":10})

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [X] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [X] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [X] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [X] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [X] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [X] Device used (e.g., Sonoff Basic): _____
- [X] Tasmota binary firmware version number used: 12.1.1
  - [X] Pre-compiled
  - [ ] Self-compiled
- [X] Flashing tools used: OTA Upgrade
- [X] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
stat/eden_switch/RESULT = {"NAME":"MJ-SD01 Dimmer Switch","GPIO":[34,33,0,323,576,322,0,0,321,416,320,96,256,0],"FLAG":0,"BASE":73}
stat/eden_switch/RESULT = {"Module":{"0":"MJ-SD01 Dimmer Switch"}}
stat/eden_switch/RESULT = {"GPIO0":{"34":"Button3"},"GPIO1":{"33":"Button2"},"GPIO2":{"0":"None"},"GPIO3":{"323":"Led_i4"},"GPIO4":{"576":"LedLink_i"},"GPIO5":{"322":"Led_i3"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"321":"Led_i2"},"GPIO13":{"416":"PWM1"},"GPIO14":{"320":"Led_i1"},"GPIO15":{"96":"Button_i1"},"GPIO16":{"256":"Relay_i1"},"GPIO17":{"0":"None"}}

```
- [X] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
stat/eden_switch/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
stat/eden_switch/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
stat/eden_switch/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}

```
- [X] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
stat/eden_switch/STATUS = {"Status":{"Module":0,"DeviceName":"Switch - Eden","FriendlyName":["Switch - Eden"],"Topic":"eden_switch","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":0,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
stat/eden_switch/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:25:14","StartupUTC":"2022-09-15T21:25:24","Sleep":50,"CfgHolder":4617,"BootCount":181,"BCResetTime":"2022-09-14T20:26:32","SaveCount":719,"SaveAddress":"F4000"}}
stat/eden_switch/STATUS2 = {"StatusFWR":{"Version":"12.1.1(tasmota)","BuildDateTime":"2022-08-25T11:33:55","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"382/699"}}
stat/eden_switch/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["ftwHome",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000080","00006010","00004000","00000000"]}}
stat/eden_switch/STATUS4 = {"StatusMEM":{"ProgramSize":624,"Free":376,"Heap":22,"ProgramFlashSize":1024,"FlashSize":4096,"FlashChipId":"16405E","FlashFrequency":40,"FlashMode":3,"Features":["00000809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000000"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62","Sensors":"1,2,3,4,5,6"}}
stat/eden_switch/STATUS5 = {"StatusNET":{"Hostname":"eden-switch-4894","IPAddress":"192.168.14.80","Gateway":"192.168.15.1","Subnetmask":"255.255.254.0","DNSServer1":"192.168.15.21","DNSServer2":"0.0.0.0","Mac":"48:55:19:15:73:1E","Webserver":2,"HTTP_API":1,"WifiConfig":2,"WifiPower":17.0}}
stat/eden_switch/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.15.38","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_15731E","MqttUser":"mqtt_user","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
stat/eden_switch/STATUS7 = {"StatusTIM":{"UTC":"2022-09-15T21:50:38","Local":"2022-09-15T22:50:38","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"06:27","Sunset":"19:03"}}
stat/eden_switch/STATUS10 = {"StatusSNS":{"Time":"2022-09-15T22:50:38"}}
stat/eden_switch/STATUS11 = {"StatusSTS":{"Time":"2022-09-15T22:50:38","Uptime":"0T00:25:14","UptimeSec":1514,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Dimmer":20,"Fade":"OFF","Speed":1,"LedTable":"ON","Wifi":{"AP":1,"SSId":"ftwHome","BSSId":"70:A7:41:6A:3C:9E","Channel":6,"Mode":"11n","RSSI":68,"Signal":-66,"LinkCount":1,"Downtime":"0T00:00:03"}}}

```
- [X] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
(Log shows me turn on the switch, press up (button 3), hold up, press down (button 2) and hold down)
22:52:05.308 MQT: stat/eden_switch/RESULT = {"POWER":"ON"}
22:52:05.313 MQT: stat/eden_switch/POWER = ON
22:52:06.109 CFG: Saved to flash at FA, Count 721, Bytes 4096
22:52:06.118 SRC: MQTT
22:52:06.121 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 17, Pld -99, Data 'NoDelay;Power1 ON'
22:52:06.126 SRC: Backlog
22:52:06.129 CMD: Grp 0, Cmd 'POWER', Idx 1, Len 2, Pld 1, Data 'ON'
22:52:06.134 MQT: stat/eden_switch/RESULT = {"POWER":"ON"}
22:52:06.137 MQT: stat/eden_switch/POWER = ON
22:52:06.280 WIF: Checking connection...
22:52:07.260 APP: Button3 multi-press 1
22:52:08.105 CFG: Saved to flash at F9, Count 722, Bytes 4096
22:52:13.654 APP: Button3 multi-press 1
22:52:15.103 CFG: Saved to flash at F8, Count 723, Bytes 4096
22:52:16.883 APP: Button3 multi-press 1
22:52:18.127 CFG: Saved to flash at F7, Count 724, Bytes 4096
22:52:19.108 CFG: Saved to flash at F6, Count 725, Bytes 4096
22:52:23.537 APP: Button2 multi-press 1
22:52:25.101 CFG: Saved to flash at F5, Count 726, Bytes 4096
22:52:26.306 APP: Button2 multi-press 1
22:52:26.309 WIF: Checking connection...
22:52:26.403 WIF: Sending Gratuitous ARP
22:52:27.205 CFG: Saved to flash at F4, Count 727, Bytes 4096

```

### TO REPRODUCE
_Steps to reproduce the behavior:_
Purchase and install Martin Jerry SD01 Dimmer (MJ-SD01), connect to network, connect to MQTT server.  Press dimmer buttons

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
I am expecting that when I press the dimmer button is results in the dimmer value on the web interface to change and for it to send the new dimmer value via an MQTT message.  The physical Dimmer Status LEDs change, but the Dimmer slider on the web interface does not.

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by anton-v-a** [comment]
I have the same problem. Is there a solution?

**Reply by abishur** [reply]
Actually yes!  I forgot I opened this ticket by the time I finally found a solution.  

The Button presses don't send out messages specifically even though it changes the dimmer value.  On the tasmota console do:

`SO59 1`

This sets option 59 to true and it will force Tasmota to send out updates anytime anything light based changes.  It should be on by default, in my opinion, but it's not.

Here's the full description for SetOption59:

```
Send tele/%topic%/STATE in addition to stat/%topic%/RESULT for commands: State, Power, and any command causing a light to be turned on.
0 = disable (default)
1 = enable
```
